### DISCUSSION TITLE
Envisager le markdown comme syntaxe de base de Publicodes

### DISCUSSION THREAD

**Discussion by mquandalle** [publicodes/publicodes#60] [Ideas]
Discussion démarrée dans https://github.com/betagouv/mon-entreprise/issues/1313#issuecomment-759459908, je crée un ticket dédié car c'est hors sujet pour betagouv/publicodes#29.

**Comment by laem** [comment]
Cette idée colle bien aussi avec le besoin de faire des "rapports" sur certains périmètres. Par exemple, dans tout le sujet climat, on pourrait faire une, 2, plusieurs pages sur le vélo. 

On n'a pas vraiment moyen aujourd'hui de l'afficher comme une page Web à scroller. Contrairement à l'implémentation actuelle, en petits liens, on perd l'idée fondamentale d'un rapport : je sais avant de le lire qu'il y a x pages de contenu. 

Les pages wikipedia sont un bon intermédiaire : je sais que la page "vélo" fait des dizaines de pages, mais j'ai aussi plein de liens hypertextes vers l'intérieur du document comme vers l'extérieur. https://fr.wikipedia.org/wiki/Bicyclette

**Comment by denismerigoux** [comment]
Pour info, Catala est passé en syntaxe markdown en mars : https://github.com/CatalaLang/catala/pull/84. Voilà ce que ça donne : https://github.com/CatalaLang/catala/blob/master/examples/allocations_familiales/securite_sociale_D.catala_fr.

**Comment by laem** [comment]
Merci ! Je me pose du coup la question de ce que je mettrais dans les paragraphes markdown entre les extraits de code. 
Ce serait plus ou moins ce que je mets dans les attributs `titre, description et note`. Sauf que ça devient du coup ambigu : à quelle règle est-ce rattaché, celle d'avant ou d'après ?

**Comment by denismerigoux** [comment]
En Catala, c'est rattaché à un "article" de loi qui est un header spécial. Voilà l'algorithme qui prend une liste de headers, de texte et de code et qui le transforme en arbre structuré :

https://github.com/CatalaLang/catala/blob/2cad9823f74ebc2fb13dd90a061e9bb4fbcb4da1/src/catala/surface/parser_driver.ml#L63
